### PR TITLE
Improve readability and remove duplication in ItemsetBinding.initReferences; fix @link’s

### PR DESCRIPTION
**Pull request by dcbriccetti**
Stepping through the code in the debugger, I needed to clean up ItemsetBinding.initReferences in order to understand it. If the improvement is useful, let’s merge it.

I also found some doc comment `link`s missing the `@`.

#### What has been done to verify that this works as intended?
Changes are made with safe IDEA tools like extract variable and method.

### CODE REVIEW

File: src/org/javarosa/core/model/ItemsetBinding.java [hunk 1]
```diff
@@ -114,32 +114,39 @@ public IConditionExpr getRelativeValue () {
...
+    /** Returns a TreeReference from an absolute reference from an IConditionExpr and a parent TreeReference */
+    private static TreeReference getCondAbsTreeRef(IConditionExpr condExpr, TreeReference parentTreeRef) {
+        XPathPathExpr xPathPathExpr = getCastExpr(condExpr);
+        return getPathAbsTreeRef(new XPathReference(xPathPathExpr), parentTreeRef);
+    }
+
+    /** Returns a TreeReference from an absolute reference from an XPathReference and a parent TreeReference */
+    private static TreeReference getPathAbsTreeRef(XPathReference xPathReference, TreeReference parentTreeRef) {
+        IDataReference dataReference = getAbsRef(xPathReference, parentTreeRef);
+        return (TreeReference) dataReference.getReference();
```
Comment (dcbriccetti) [inline src/org/javarosa/core/model/ItemsetBinding.java]: I inlined FormInstance.unpackReference (which doesn’t provide any value here):
```
    public static TreeReference unpackReference(IDataReference ref) {
        return (TreeReference) ref.getReference();
    }
```

**Comment by dcbriccetti** [comment]
Thanks, @lognaturel. I think I made the changes you recommended, and I added your explanatory text.

File: src/org/javarosa/core/model/ItemsetBinding.java [hunk 2]
```diff
@@ -114,32 +114,34 @@ public IConditionExpr getRelativeValue () {
...
-                    nodesetRef));
-        }
-        if ( valueExpr != null ) {
-            valueRef = FormInstance.unpackReference(FormDef.getAbsRef(new XPathReference(((XPathPathExpr) ((XPathConditional) valueExpr).getExpr())),
-                    nodesetRef));
-        }
 
-        if ( q != null ) {
+        // Convert the expression to a relative reference and then anchor it to the context.
+        nodesetRef = (TreeReference) getAbsRef(new XPathReference(
```
Comment (lognaturel) [inline src/org/javarosa/core/model/ItemsetBinding.java]: You should be able to just use `getAbsoluteRef(nodesetExpr, contextRef)`, here. Did that not work?

**Comment by dcbriccetti** [comment]
I didn’t completely follow the noop comment (we can move on), but I’ll trust you that it works. I made the change. Thanks.

**Comment by lognaturel** [comment]
Trust is a dangerous thing!

The original expression was `new XPathReference(((XPathPathExpr) ((XPathConditional) nodesetExpr).getExpr()).getReference())`. `XPathPathExpr.getReference()` does a whole bunch of stuff to go from a path expression to a reference. Then the `XPathReference(XPathReference)` constructor was called with it and that basically makes a copy of the parameter (no-op). 

For the others, `.getReference()` wasn't called explicitly. That means the `XPathReference(XPathPathExpr)` constructor was being called. That constructor calls `.getReference()` on its parameter. That's the same thing that was happening above.

**Pull request merged**